apps/qubit/modules/informationobject/actions/reportsAction.class.php

```diff
@@ -25,6 +25,70 @@ class InformationObjectReportsAction extends sfAction
       'report'
     );
 
+  public function execute($request)
+  {
+    $this->typeLabels = array(
+      'fileList' => $this->context->i18n->__('File list'),
+      'itemList' => $this->context->i18n->__('Item list'),
+      'storageLocations' => $this->context->i18n->__('Physical storage locations'),
+      'boxLabel' => $this->context->i18n->__('Box label')
+    );
+
+    $this->resource = $this->getRoute()->resource;
+    $this->getExistingReports();
+
+    if (!isset($this->resource))
+    {
+      $this->forward404();
+    }
+
+    $this->form = new sfForm;
+
+    foreach ($this::$NAMES as $name)
+    {
+      $this->addField($name);
+    }
+
+    if ($request->isMethod('post'))
+    {
+      $this->form->bind($request->getPostParameters());
+      if ($this->form->isValid())
+      {
+        $this->redirect($this->form->getValue('report'));
+      }
+    }
+  }
+
+  private function getExistingReports()
+  {
+    $formats = array('csv', 'html');
+    $types = array_keys($this->typeLabels);
+    $this->existingReports = array();
+
+    foreach ($types as $type)
+    {
+      foreach ($formats as $format)
+      {
+        $path = arGenerateReportJob::getFilename($this->resource, $format, $type);
+
+        if (file_exists($path))
+        {
+          if (!sfContext::getInstance()->user->isAuthenticated() &&
+              in_array($type, array('storageLocations', 'boxLabel')))
+          {
+            continue;
+          }
+
+          $this->existingReports[] = array(
+            'path' => sfConfig::get('siteBaseUrl').DIRECTORY_SEPARATOR.$path,
```

Comment:
I agree, I read an article after saying it's pointless unless you're using it to explode a path a system call passes back to you. Missed this one!